src/main/webapp/app/exam/participate/exam-participation.component.ts

```diff
@@ -77,7 +77,14 @@ export class ExamParticipationComponent implements OnInit, OnDestroy {
         this.route.parent!.params.subscribe((params) => {
             this.courseId = parseInt(params['courseId'], 10);
             this.examId = parseInt(params['examId'], 10);
-            this.examParticipationService.loadExam(this.courseId, this.examId).subscribe((exam) => (this.exam = exam));
+            this.examParticipationService.loadExam(this.courseId, this.examId).subscribe((exam) => {
+                this.exam = exam;
+                if (this.isOver()) {
+                    this.examParticipationService.loadStudentExam(this.exam.course.id, this.exam.id).subscribe((studentExam: StudentExam) => {
+                        this.studentExam = studentExam;
```

Comment:
I will approve this for now, but in the future we should check if the student has actually participated in the exam and only then we will show the summary page. We might need to add a new attribute to StudentExam to track this. (something similar like the initializationState in the Participation)